docs/reference/indices/put-mapping.asciidoc

```diff
@@ -166,35 +158,430 @@ PUT /my_index <1>
-<2> Add a `last` field under the `name` object field.
-<3> Update the `ignore_above` setting from its default of 0.
+The API returns the following response:
 
-Each <<mapping-params,mapping parameter>> specifies whether or not its setting
-can be updated on an existing field.
+[source,console-result]
+----
+{
+  "my_index" : {
+    "mappings" : {
+      "properties" : {
+        "user_id" : {
+          "type" : "keyword",
+          "ignore_above" : 100
+        }
+      }
+    }
+  }
+}
+----
+
+
+[[updating-field-mappings]]
+===== Change the mapping of an existing field
+
+// tag::change-field-mapping[]
+Except for supported <<mapping-params,mapping parameters>>,
+you can't change the mapping or field type of an existing field.
+Changing an existing field could invalidate data that's already indexed.
+
+If you need to change the mapping of a field,
+create a new index with the correct mapping
+and <<docs-reindex,reindex>> your data into that index.
+// end::change-field-mapping[]
+
+To see how this works,
+try the following example.
+
+Use the <<indices-create-index,create index>> API
+to create the `users` index
+with the `user_id` field
+with the <<number,`long`>> field type.
+
+[source,console]
+----
+PUT /users
+{
+  "mappings" : {
+    "properties": {
+      "user_id": {
+        "type": "long"
+      }
+    }
+  }
+}
+----
+
+Use the <<docs-index_,index>> API
+to index several documents
+with `user_id` field values.
+
+[source,console]
+----
+POST /users/_doc?refresh=wait_for
+{
+    "user_id" : 12345
+}
+
+POST /users/_doc?refresh=wait_for
+{
+    "user_id" : 12346
+}
+----
+// TEST[continued]
+
+To change the `user_id` field
+to the <<keyword,`keyword`>> field type,
+use the create index API
+to create the `new_users` index with the correct mapping.
+
+[source,console]
+----
+PUT /new_users
+{
+  "mappings" : {
+    "properties": {
+      "user_id": {
+        "type": "keyword"
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+Use the <<docs-reindex,reindex>> API
+to copy documents from the `users` index
+to the `new_users` index.
+
+[source,console]
+----
+POST /_reindex
+{
+  "source": {
+    "index": "users"
+  },
+  "dest": {
+    "index": "new_users"
+  }
+}
+----
+// TEST[continued]
+
+The API returns the following response:
+
+[source,console-result]
+----
+{
+  "took": 147,
+  "timed_out": false,
+  "total": 2,
+  "updated": 0,
+  "created": 2,
+  "deleted": 0,
+  "batches": 1,
+  "version_conflicts": 0,
+  "noops": 0,
+  "retries": {
+    "bulk": 0,
+    "search": 0
+  },
+  "throttled_millis": 0,
+  "requests_per_second": -1.0,
+  "throttled_until_millis": 0,
+  "failures" : [ ]
+}
+----
+// TESTRESPONSE[s/"took": 147/"took": "$body.took"/]
+
+
+[[rename-existing-field]]
+===== Rename a field
+
+// tag::rename-field[]
+If you want to rename a field,
+consider adding an <<alias, `alias`>> field to the index.
+// end::rename-field[]
+
```

Comment:
Should this say something about why you can't just rename the field? Something like:

Renaming a field would invalidate data already indexed under the old field name. 
Instead, add an alias field to create an alternate field name.

Reply:
Thanks @debadair. Added with 852a9e68aba.